youtube_dl/extractor/megaphone.py

```diff
@@ -45,11 +68,225 @@ def _real_extract(self, url):
+                self.to_screen('Downloading just the first episode because of --no-playlist')
+                return entries['entries'][0]
+        return entries
+
+
+class MegaphoneChannelIE(MegaphoneIE):
+    IE_NAME = 'megaphone.fm:channel'
+    IE_DESC = 'megaphone.fm channel'
+    _VALID_URL = r'https://cms\.megaphone\.fm/channel/(?P<id>[A-Z0-9]+)(?:\?selected=(?P<clip_id>[A-Z0-9]+))?'
+    _TESTS = [{
+        'url': 'https://cms.megaphone.fm/channel/ADL3707263633',
+        'info_dict': {
+            'id': 'ADL3707263633',
+            'title': 'Pax Britannica',
+            'description': 'md5:7b4002330ffe4abcb81d97ab9b56fede',
+        },
+        'playlist_mincount': 98,
+    }, {
+        'url': 'https://cms.megaphone.fm/channel/ADL3707263633?selected=ADL9449136081',
+        'file_minsize': 29000000,
+        'info_dict': {
+            'id': 'ADL9449136081',
+            'title': '02.23 - Nolumus Leges Angliae Mutari',
+            'description': 'md5:d35989ec81de7199b3020bc919ab7c0d',
+            'ext': 'mp3',
+            'thumbnail': r're:^https://.*\.png(?:\?.+)?$',
+            'duration': 2470.09,
+            'extractor_key': 'Megaphone',
+            'upload_date': '20210711',
+            'timestamp': 1625961600,
+            'uploader': 'Pax Britannica',
+        },
+    }, {
+        'url': 'https://cms.megaphone.fm/channel/ADL3707263633',
+        'file_minsize': 25000000,
+        'info_dict': {
+            'id': 'ADL9716153485',
+            'title': '02.24 - Give Unto Caesar His Due',
+            'description': 'The First English Civil War begins.',
+            'ext': 'mp3',
+            'duration': 1860,
+        },
+        'params': {
+            'noplaylist': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url).groupdict()
+        video_id = mobj['id']
+        # If clip is selected, dl that instead
+        clip_id = mobj.get('clip_id')
+        if clip_id:
+            return self.url_result(self._PLAYER_URL_TEMPL % clip_id, ie='Megaphone', video_id=clip_id)
+        webpage = self._download_webpage(url, video_id)
+        # Useful information is split between a JS JSON.parse() call and
+        # a <div class="public-ep-list>...</div> HTML element
+        playlist_json =  self._search_regex(
+	        r"var\s+playlist\s*=\s*JSON\.parse\s*\(\s*('[^<]+?')\s*\)\s*;",
+                webpage, 'playlist JSON')
+        # The data must be parsed twice: once for JSON.parse, once for Python
+        playlist_json = self._parse_json(
+            playlist_json, video_id, 
+            transform_source=lambda s: self._parse_json(s, video_id, transform_source=js_to_json))
+        entries = []
+        ep_num = 0
+        ep_list = None
+        # Support --no-playlist to get the first item when no explicit selection
+        noplaylist = self._downloader.params.get('noplaylist')
+        for episode in playlist_json:
+            ep_num += 1
```

Comment:
Would using `for ep_num, episode in enumerate(playlist_json, 1)` and later `if len(...)` or simply `if:` make more sense since you'll be counting the episodes ?

Reply:
Yes.